docs/DMS/PRIMARY-DATA-STORAGE.md

```diff
@@ -25,3 +25,268 @@ Braindump:
+
+### Option C: One table per resource, single reference table
+
+Option C is like Option B except it has the single reference table like Option A to avoid the problem of join tables.
+
+```mermaid
+erDiagram
+    StudentSchoolAssociation {
+        Guid referential_id PK "Hash of identity"
+        Guid document_uuid "API resource id, indexed"
+        Guid student_referential_id FK "Identity of Student"
+        Guid school_referential_id FK "Identity of School"
+        JSON edfi_doc "The document"
+        string graduation_school_year "Example query field"
+        string entry_date "Example query field"
+        string et_cetera
+    }
+    References }o--|| StudentSchoolAssociation : ""
+    References }o--|| GraduationPlan : ""
+    References {
+        Guid parent_ref_id "Parent document with reference - not an FK constraint"
+        Guid referenced_ref_id  "Document being referenced - not an FK constraint"
+    }
+    GraduationPlan {
+        Guid referential_id PK "Hash of identity"
+        Guid document_uuid "API resource id, indexed"
+        JSON edfi_doc "The document"
+        string et_cetera
+    }
+```
+
+The problem is that with a single `References` table, we cannot rely on referential integrity to do reference validation. This means that we have to manage referential integrity entirely with Tanager. Our experience with this in Meadowlark showed a couple issues. The most important is that it is error-prone and easy to get wrong. The second is that while Meadowlark was able to get this down to simple index lookups, it still required several round trips to accomplish reference validation.
+
+We could mitigate the round trips by pushing reference validation into stored procedures. However, addressing the potential for errors requires a very heavy investment in testing. This would be even more complex if we had to do testing on stored procedures.
+
+
+## Proposed Proof of Concept for Option A
+
+Option A is our preferred alternative. A plan to test it:
+
+- Create `Documents` and `References` with reference IDs as uuids for the primary key
+- Insert a million records to start. (We'd really like to get more like 100 million in order to get the right order of magnitude for large districts, but a test like that takes a long time.)
+  - Monitor index fragmentation with the tools from [this presentation](https://www.youtube.com/watch?v=nc4CMo7VSPo), and do index maintenance as necessary
+- Compare insert performance.
+- Compare query performance with a couple of likely scenarios.
+- Compare storage requirements.
+- Repeat with bigints as primary keys with sequential ordering
```

Comment:
This isn't trivial right? Won't we have to modify a lot of the Tanager code to run this leg of the experiment?

Reply:
We can probably do some pure SQL experimentation or some throw away C# code. I still need to review this section of document.

Reply:
Agree this needs more detail. Intent is for the POC to be a simulation of how Tanager would behave under these scenarios.